cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/MessageBatchQueryProvider.java

```diff
@@ -224,24 +218,56 @@ private BoundStatementBuilder bindFilterParameters(BoundStatementBuilder builder
 			builder = builder.setLong(RIGHT_MESSAGE_INDEX, batch != null ? batch.getMessageIndex() : filter.getIndex().getValue());
 		}
 		
-		//Both filters for timestamp are adjusted to get more batches than we will get in case of strict comparison.
-		//This is to cover the case when messages that meet condition timestamp are in the middle of the batch and the batch is long in terms of time.
-		//E.g. in batch first_message_time=15:00:00, last_message_time=15:05:00, 
-		//condition timestampFrom>15:01:00
-		//Without condition adjustment we'll skip such batch thus not showing messages that actually met the condition
-		//FIXME: This doesn't guarantee that batches longer than 10 minutes will be not skipped! Anyway, such batches are bad.
 		if (filter.getTimestampFrom() != null)
 		{
 			Instant ts = filter.getTimestampFrom().getValue();
-			builder = FilterUtils.bindTimestamp(ts.minus(10, ChronoUnit.MINUTES), builder, FIRST_MESSAGE_DATE, FIRST_MESSAGE_TIME);
+			try
+			{
+				long leftBatchIndex = getNearestMessageIndexBefore(tmOperator, instanceId,
+						filter.getStreamName().getValue(), directionFilter.getValue(), ts, attributes);
+				builder = builder.setLong(LEFT_MESSAGE_INDEX, leftBatchIndex);
```

Comment:
There seem to be a conflict if someone defines both timestampFrom and index in the filter. Better use different placeholder name for index obtained for timestampFrom.